### PR TITLE
Adding SC wrapper

### PR DESCRIPTION
**Pull request by yger**

### CODE REVIEW

File: neo/rawio/spykingcircusrawio.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        BaseRawIO.__init__(self)
+        self.dirname = dirname
+
+    def _source_name(self):
+        return self.dirname
+
+    def _parse_header(self):
+        assert HAVE_HDF5, self.installation_mes
+        spykingcircus_folder = Path(self.dirname)
+        SpykingCircusRawIO.__init__(self, spykingcircus_folder)
```
Comment (samuelgarcia) [inline neo/rawio/spykingcircusrawio.py:49]: ????

File: neo/rawio/spykingcircusrawio.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            if 'result-merged.hdf5' in str(f):
+                results = f
+                break
+
+        # load params
+        for f in parent_folder.iterdir():
+            if f.suffix == '.params':
+                sample_rate = _load_sample_rate(f)
+
+        if sample_rate is not None:
```
Comment (samuelgarcia) [inline neo/rawio/spykingcircusrawio.py:84]: Shoudl raise an error if it is None

File: neo/rawio/spykingcircusrawio.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+        if results is None:
+            raise Exception(spykingcircus_folder, " is not a spyking circus folder")
+        f_results = h5py.File(results, 'r')
+
+        self._all_spikes = []
+        self.unit_labels = []
+        for temp in f_results['spiketimes'].keys():
+            self._all_spikes.append(np.array(f_results['spiketimes'][temp]).astype('int64'))
+            self.unit_labels.append(int(temp.split('_')[-1]))
```
Comment (samuelgarcia) [inline neo/rawio/spykingcircusrawio.py:95]: You should not load all then now but on demand

File: neo/rawio/spykingcircusrawio.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        event_channels = np.array(event_channels, dtype=_event_channel_dtype)
+
+        self.header = {}
+        self.header['nb_block'] = 1
+        self.header['nb_segment'] = [1]
+        self.header['signal_channels'] = sig_channels
+        self.header['unit_channels'] = unit_channels
+        self.header['event_channels'] = event_channels
+
+        self.duration = f_results['info']['duration'][0]
```
Comment (samuelgarcia) [inline neo/rawio/spykingcircusrawio.py:125]: Maybe a private attribute

File: neo/rawio/spykingcircusrawio.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+
+    def _get_signal_t_start(self, block_index, seg_index, channel_indexes):
+        return None
+
+    def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop, channel_indexes):
+        return None
+
+    def _spike_count(self, block_index, seg_index, unit_index):
+        unit_label = self.unit_labels[unit_index]
+        nb_spikes = np.sum(self._all_spikes[unit_label])
```
Comment (samuelgarcia) [inline neo/rawio/spykingcircusrawio.py:146]: This is wrong

File: neo/rawio/spykingcircusrawio.py [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+        return nb_spikes
+
+    def _get_spike_timestamps(self, block_index, seg_index, unit_index, t_start, t_stop):
+        assert block_index == 0
+        assert seg_index == 0
+
+        unit_label = self.unit_labels[unit_index]
```
Comment (samuelgarcia) [inline neo/rawio/spykingcircusrawio.py:153]: Remove this label.

File: neo/rawio/spykingcircusrawio.py [hunk 7]
```diff
@@ -0,0 +1,183 @@
+"""
+
```
Comment (samuelgarcia) [inline neo/rawio/spykingcircusrawio.py:2]: Add some comments here to dscribe the reader and link to the project.

File: neo/rawio/spykingcircusrawio.py [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+                sample_rate = r.split('=')[-1]
+                if '#' in sample_rate:
+                    sample_rate = sample_rate[:sample_rate.find('#')]
+                sample_rate = float(sample_rate)
+    return sample_rate
+
+
+class SpykingCircusRawIO(BaseRawIO):
+    """
+
```
Comment (samuelgarcia) [inline neo/rawio/spykingcircusrawio.py:34]: Short doc here

File: neo/rawio/spykingcircusrawio.py (hunk 1 above)
Comment (yger) [reply neo/rawio/spykingcircusrawio.py]: Commented

File: neo/rawio/spykingcircusrawio.py [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+    def __init__(self, dirname=''):
+        BaseRawIO.__init__(self)
+        self.dirname = dirname
+
+    def _source_name(self):
+        return self.dirname
+
+    def _parse_header(self):
+        assert HAVE_HDF5, self.installation_mes
```
Comment (samuelgarcia) [inline neo/rawio/spykingcircusrawio.py:47]: remove self.installation_mes

File: neo/rawio/spykingcircusrawio.py [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+        spykingcircus_folder = Path(self.dirname)
+        #SpykingCircusRawIO.__init__(self, spykingcircus_folder)
```
Comment (samuelgarcia) [inline neo/rawio/spykingcircusrawio.py:49]: remove this

File: neo/rawio/spykingcircusrawio.py (hunk 2 above)
Comment (yger) [reply neo/rawio/spykingcircusrawio.py:84]: Raised an error

File: neo/rawio/spykingcircusrawio.py (hunk 3 above)
Comment (yger) [reply neo/rawio/spykingcircusrawio.py]: Only unit_index now

File: neo/rawio/spykingcircusrawio.py (hunk 5 above)
Comment (yger) [reply neo/rawio/spykingcircusrawio.py]: done

File: neo/rawio/spykingcircusrawio.py (hunk 6 above)
Comment (yger) [reply neo/rawio/spykingcircusrawio.py]: done

**Comment by samuelgarcia** [comment]
To have tests passing you maybe have to play with this installtion file:
circleci/requirements_testing.txt

**Comment by samuelgarcia** [comment]
Hi Pierre. Could fix pep8 and small comments so I could merge this ?

**Comment by samuelgarcia** [comment]
Hi Pierre.
Would you time time to finish this for 0.10 release ?
need pass test and pep8 stuff.

**Comment by yger** [comment]
Back from holidays, I'll try to finish that asap

File: neo/rawio/spykingcircusrawio.py (hunk 9 above)
Comment (yger) [reply neo/rawio/spykingcircusrawio.py:47]: Done

File: neo/rawio/spykingcircusrawio.py (hunk 10 above)
Comment (yger) [reply neo/rawio/spykingcircusrawio.py:49]: Done

**Comment by JuliaSprenger** [comment]
@yger Is this PR still alive? I created a rebased version of this branch [here](https://github.com/JuliaSprenger/python-neo/tree/spiking_circus_IO) in case the rebasing is the issue.

**Pull request closed without merging**